server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -330,7 +332,14 @@ public ClusterState execute(ClusterState currentState) {
 
                         shards = shardsBuilder.build();
                         RestoreInProgress.Entry restoreEntry = new RestoreInProgress.Entry(snapshot, overallState(RestoreInProgress.State.INIT, shards), Collections.unmodifiableList(new ArrayList<>(indices.keySet())), shards);
-                        builder.putCustom(RestoreInProgress.TYPE, new RestoreInProgress(restoreEntry));
+                        List<RestoreInProgress.Entry> newEntries;
```

Comment:
Here we did assume only a single restore in parallel previously, now we have to add `restoreEntry` to the existing list if it's there.